File: WebExam/Models/Question.cs
namespace WebExam.Models
{
    public class Question : BaseEntity
    {
        public int ExamId { get; set; }
        public string Text { get; set; }
        public QuestionType Type { get; set; }
        public int Points { get; set; }
        public int Order { get; set; }

        // Навигационные свойства
        public virtual Exam Exam { get; set; }
        public virtual ICollection<AnswerOption> AnswerOptions { get; set; }
    }

    public enum QuestionType
    {
        SingleChoice = 1,
        MultipleChoice = 2,
        TextAnswer = 3,
        CodeAnswer = 4
    }
}

File: WebExam/WebExam-Front/Models/question.ts
export interface Question {
    id: number;
    text: string;
    type: QuestionType;
    points: number;
    order: number;
    answerOptions: AnswerOption[];
}

export type QuestionType = 'SingleChoice' | 'MultipleChoice' | 'TextAnswer' | 'CodeAnswer';

export interface AnswerOption {
    id: number;
    text: string;
    isCorrect: boolean;
    order?: number;
}